### PR TITLE
Vkyfox

### PR DESCRIPTION
**Pull request by vkyfox**
[fr] rule improvements, added new rules #2900 and anglicisme

### CODE REVIEW

File: languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml [hunk 1]
```diff
@@ -17896,35 +17950,77 @@ though, and has since been slightly modified)-->
...
+              <example correction="captures d'écran">J'en ai pris un <marker>screen shots</marker> si tu veux.</example>
+          </rule>
+      </rulegroup>
+      <rulegroup id="SCREENCAST" name="Anglicisme : screencast" default="temp_off">
+          <rule>
+              <pattern>
+                  <token regexp="yes">screencasts?</token>
+              </pattern>
+              <message>« \1 » est un anglicisme. L'équivalent français est « capture(s) vidéo ».</message>
+              <suggestion><match no="1" regexp_match="(screencast)(s)?" regexp_replace="capture$2"/> vidéo</suggestion>
```
Comment (tiff) [inline languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:17980]: Will this also work when "Screencast" is at the beginning of a sentence and therefore capitalized? @vkyfox

Comment (tiff) [reply languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:17980]: Same applies to the other rules.

Comment (jaumeortola) [reply languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:17980]: @vkyfox You need to add `regexp_match="(?iu)(screencast)(s)?"` to make it case-insensitive.

Comment (vkyfox) [reply languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml]: Such does not seem to be the case in <pattern> rules as the capitalized examples I added are detected in the test runs. I suppose this has to do with the case_sensitive attribute for the token taking priority on the regexp itself?
I had to adjust the regexp in the suggestion for it to work properly, however, so thank you for bringing this to my attention!

Comment (jaumeortola) [reply languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml]: The pattern rules are case-insensitive by default. But the `regexp_match` (for generating suggestions) is case-sensitive. You can do `regexp_match="([Ss]creenshot)(s)?"` (for capitalized words) or regexp_match="(?iu)(screencast)(s)?" (fully case-insensitive).

Comment (vkyfox) [reply languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml]: I did.

**Pull request merged**